src/WinGetCommandNotFound.cs

```diff
@@ -0,0 +1,302 @@
+        {
+            var results = new List<CatalogPackage>();
+            if (matches.Count == 1)
+            {
+                // One match --> return the package
+                results.Add(matches.First().CatalogPackage);
+            }
+            else if (matches.Count > 1)
+            {
+                // Multiple matches --> display top 5 matches (prioritize best matches first)
+                var bestExactMatches = new List<CatalogPackage>();
+                var secondaryMatches = new List<CatalogPackage>();
+                var tertiaryMatches = new List<CatalogPackage>();
+                for (int i = 0; i < matches.Count; i++)
+                {
+                    var match = matches[i];
+                    switch (match.MatchCriteria.Option)
+                    {
+                        case PackageFieldMatchOption.EqualsCaseInsensitive:
+                        case PackageFieldMatchOption.Equals:
+                            bestExactMatches.Add(match.CatalogPackage);
+                            break;
+                        case PackageFieldMatchOption.StartsWithCaseInsensitive:
+                            secondaryMatches.Add(match.CatalogPackage);
+                            break;
+                        case PackageFieldMatchOption.ContainsCaseInsensitive:
+                            tertiaryMatches.Add(match.CatalogPackage);
+                            break;
+                    }
+                }
+
+                // Now return the top _maxSuggestions
+                while (results.Count < _maxSuggestions)
+                {
+                    if (bestExactMatches.Count > 0)
+                    {
+                        results.Add(bestExactMatches.First());
+                        bestExactMatches.RemoveAt(0);
+                    }
+                    else if (secondaryMatches.Count > 0)
+                    {
+                        results.Add(secondaryMatches.First());
+                        secondaryMatches.RemoveAt(0);
+                    }
+                    else if (tertiaryMatches.Count > 0)
+                    {
+                        results.Add(tertiaryMatches.First());
+                        tertiaryMatches.RemoveAt(0);
+                    }
+                    else
+                    {
+                        break;
+                    }
+                }
+            }
+            _tooManySuggestions = matches.Count > _maxSuggestions;
+            return results;
+        }
+
+        // Adapted from WinGet sample documentation: https://github.com/microsoft/winget-cli/blob/master/doc/specs/%23888%20-%20Com%20Api.md#32-search
+        private List<CatalogPackage> _FindPackages(string query)
+        {
+            // Get the package catalog
+            var catalogRef = WinGetComObjects.Singleton.packageManager.GetPredefinedPackageCatalog(PredefinedPackageCatalog.OpenWindowsCatalog);
+            var connectResult = catalogRef.Connect();

```

Comment:
> There's still an issue where the feedback provider won't work for the first 2 to 3 seconds.

Which takes more the time? The initialization of `WinGetComObjects.Singleton`, or `connect()` call?
For the `WinGetComObjects` initialization, you should be able to do it in `OnImport()`, on a thread-pool thread. In that way, you won't block the module load, but is able kick off the initialization in the background, so when the first trigger comes, the initialization maybe is already done.